cirq-core/cirq/protocols/measurement_key_protocol.py

```diff
@@ -140,13 +155,67 @@ def measurement_keys(val: Any, *, allow_decompose: bool = True) -> AbstractSet[s
     return set()
 
 
-def is_measurement(val: Any) -> bool:
-    """Determines whether or not the given value is a measurement.
+def _is_measurement_from_magic_method(val: Any) -> Optional[bool]:
+    """Uses `is_measurement` magic method to determine if this object is a measurement."""
+    getter = getattr(val, '_is_measurement_', None)
+    return NotImplemented if getter is None else getter()
+
 
-    Measurements are identified by the fact that `cirq.measurement_keys` returns
-    a non-empty result for them.
+def _is_any_measurement(
+    vals: List[Any], vals_to_decompose: List[Any], allow_decompose: bool
```

Comment:
It doesn't look like `vals_to_decompose` needs to be an arg - it could be constructed in the method.

Reply:
Good catch. It was a remnant from when I had implemented this method recursively.